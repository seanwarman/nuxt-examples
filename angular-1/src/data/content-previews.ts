import { RoomPlannerImages, SectionImages } from '../assets/images';

export interface ContentPreview {
  label: string;
  primaryImage: string;
  icon: string;
  body: string;
  slug: string;
  firstSubsection?: string;
  learnIcon: string;
  designIcon: string;
}

export const contentPreviews: ContentPreview[] = [
  {
    slug: 'camera-and-audio',
    label: 'Camera & Audio',
    firstSubsection: 'camera',
    primaryImage: SectionImages.cameraAndAudio.contentPreviewHero,
    icon: SectionImages.cameraAndAudio.icons.sectionIcon,
    designIcon: SectionImages.cameraAndAudio.icons.designIcon,
    learnIcon: SectionImages.cameraAndAudio.icons.learnIcon,
    body: `We've tuned and tweaked the hardware. Now, position it correctly to
     unlock the best performance.`,
  },
  {
    slug: 'displays-and-extras',
    label: 'Displays & Extras',
    firstSubsection: 'screen',
    primaryImage: SectionImages.displaysAndExtras.contentPreviewHero,
    icon: SectionImages.displaysAndExtras.icons.sectionIcon,
    designIcon: SectionImages.displaysAndExtras.icons.designIcon,
    learnIcon: SectionImages.displaysAndExtras.icons.learnIcon,
    body: `Correctly sizing displays is a key part of the conference experience.
      Plus, get some hints and tips on using the Series One Board 65 and Desk 27,
      developed in collaboration with Avocor,
      in your space.`,
  },
  {
    slug: 'room-layout',
    label: 'Room layout',
    firstSubsection: 'viewing-area',
    primaryImage: SectionImages.roomLayout.contentPreviewHero,
    icon: SectionImages.roomLayout.icons.sectionIcon,
    designIcon: SectionImages.roomLayout.icons.designIcon,
    learnIcon: SectionImages.roomLayout.icons.learnIcon,
    body: `Following a few spatial guidelines can make a huge difference for remote and in-room participants.`,
  },
  {
    slug: 'furniture-lighting',
    label: 'Furniture and lighting',
    firstSubsection: 'lighting',
    primaryImage: SectionImages.furnitureLighting.contentPreviewHero,
    icon: SectionImages.furnitureLighting.icons.sectionIcon,
    designIcon: SectionImages.furnitureLighting.icons.designIcon,
    learnIcon: SectionImages.furnitureLighting.icons.learnIcon,
    body: `Make your meeting room look and feel inviting with simple decorative and lighting tips. Appropriate lighting helps both local and remote participants have a great experience.`,
  },
  {
    slug: 'acoustics',
    label: 'Acoustics',
    firstSubsection: 'separation',
    primaryImage: SectionImages.acoustics.contentPreviewHero,
    icon: SectionImages.acoustics.icons.sectionIcon,
    designIcon: SectionImages.acoustics.icons.designIcon,
    learnIcon: SectionImages.acoustics.icons.learnIcon,
    body: `Poor acoustics is a common downfall of video conferencing spaces. The right allowance of acoustic finishes can help make speech clear and audible, and adequate isolation helps avoid interruptions.`,
  },
  {
    slug: 'services',
    label: 'Services',
    firstSubsection: 'air-handling',
    primaryImage: SectionImages.services.contentPreviewHero,
    icon: SectionImages.services.icons.sectionIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Make sure power is provisioned for the right places, and ensure ventilation systems don't cause unwanted noise pickup on your microphones.`,
  },
  {
    slug: 'room-planner',
    label: 'Room Designer',
    primaryImage: RoomPlannerImages.contentPreviewHero,
    icon: RoomPlannerImages.icons.mainIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Need to size up your room? All the best practices in this guide are used to help you pick the equipment and recommended room size for the occupancy you need.`,
  },
];
