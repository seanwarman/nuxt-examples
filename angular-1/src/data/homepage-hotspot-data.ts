export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  cameraPosition: {
    x: number;
    y: number;
    z: number;
  };
  cameraRotation: {
    x: number;
    y: number;
    z: number;
  };
  modelsOn: Array<string>;
  modelsOff: Array<string>;
}> = [
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -5,
    y: 113,
    z: -160,
    cameraPosition: {
      x: -100,
      y: 175,
      z: 350,
    },
    cameraRotation: {
      x: 100,
      y: 100,
      z: 0,
    },
    modelsOn: ['CameraAndAudio', 'defaultChairs', 'acousticDefault'],
    modelsOff: ['defaultCameraAndAudio'],
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -230,
    y: 150,
    z: 50,
    cameraPosition: {
      x: 100,
      y: 200,
      z: 500,
    },
    cameraRotation: {
      x: 50,
      y: 100,
      z: 0,
    },
    modelsOn: [
      'DisplaysAndExtras',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: 90,
    y: 20,
    z: 200,
    cameraPosition: {
      x: 0,
      y: 280,
      z: 800,
    },
    cameraRotation: {
      x: 150,
      y: 100,
      z: 100,
    },
    modelsOn: [
      'RoomLayout',
      'defaultChairs',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: 110,
    y: 70,
    z: 80,
    cameraPosition: {
      x: 400,
      y: 200,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 70,
      z: -100,
    },
    modelsOn: [
      'FurnitureAndLighting',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['defaultChairs'],
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -230,
    y: 200,
    z: 180,
    cameraPosition: {
      x: 300,
      y: 800,
      z: -150,
    },
    cameraRotation: {
      x: -150,
      y: 0,
      z: -100,
    },
    modelsOn: [
      'Acoustics',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticPanel00',
      'acousticPanel01',
      'acousticPanel02',
      'acousticPanel03',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['stripLight00', 'stripLight01', 'acousticDefault'],
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 110,
    y: 115,
    z: -160,
    cameraPosition: {
      x: 300,
      y: 160,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 50,
      z: -250,
    },
    modelsOn: [
      'services',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-planner',
    text: 'Room Designer',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-designer-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    cameraPosition: {
      x: -30,
      y: 150,
      z: 150,
    },
    cameraRotation: {
      x: 40,
      y: 60,
      z: 0,
    },
    modelsOn: [
      'Drawing',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
];
