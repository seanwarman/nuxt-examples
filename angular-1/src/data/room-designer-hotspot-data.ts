export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  camX: number;
  camY: number;
  camZ: number;
}> = [
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: -110,
    y: 20,
    z: 100,
    camX: -800,
    camY: 800,
    camZ: 800,
  },
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    camX: -800,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: 0,
    y: 110,
    z: -160,
    camX: -1200,
    camY: 500,
    camZ: 600,
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 120,
    y: 200,
    z: -160,
    camX: -120,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: -70,
    y: 265,
    z: 20,
    camX: -120,
    camY: 300,
    camZ: 400,
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -210,
    y: 190,
    z: -160,
    camX: -120,
    camY: 220,
    camZ: 580,
  },
  // {
  //   name: 'room-planner',
  //   text: 'Room Planner',
  //   color: '#0d9cab',
  //   defaultIcon: 'icon-hotspot-teal-01@3x.png',
  //   activeIcon: 'icon-room-designer-white-teal-circle-01@3x.png',
  //   halo: 'teal.png',
  //   x: -10,
  //   y: 150,
  //   z: -160,
  //   camX: 0,
  //   camY: 200,
  //   camZ: 200,
  // },
];
