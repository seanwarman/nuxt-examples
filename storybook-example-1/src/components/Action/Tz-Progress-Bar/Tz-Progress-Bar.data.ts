import type { ProgressBarStep } from './Tz-Progress-Bar.types';

export const exampleSteps = [
  {
    label: 'Your details',
    url: '/details'
  },
  {
    label: 'Delivery',
    url: '/delivery'
  },
  {
    label: 'Payment',
    url: '/payment'
  }
] as Array<ProgressBarStep>;
