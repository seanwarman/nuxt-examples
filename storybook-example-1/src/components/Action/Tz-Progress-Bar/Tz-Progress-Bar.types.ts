export type ProgressBarStep = {
  url: string;
  label: string;
};
