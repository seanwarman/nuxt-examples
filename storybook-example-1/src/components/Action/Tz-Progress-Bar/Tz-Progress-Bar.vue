<template>
  <div
    class="m-auto w-full pb-2"
    :class="[computedSteps.length === 3 ? 'max-w-[395px]' : steps.length > 3 && 'max-w-[800px]']"
  >
    <ul class="flex w-full">
      <li v-for="(step, k) in computedSteps" :key="k" class="relative flex-1" :class="[step.active && 'active']">
        <router-link
          v-if="step.enabled"
          :to="step.url"
          class="inline-block h-full w-full p-2 text-center text-xs uppercase tracking-widest text-gray-500 after:absolute after:top-full after:left-0 after:right-0 after:bottom-auto after:h-2 after:bg-gray-100 after:content-[''] hover:text-[#2F2F2F] hover:underline focus:text-[#2F2F2F] focus:underline focus:outline-0"
          >{{ step.label }}</router-link
        >
        <span
          v-else
          class="inline-block h-full w-full p-2 text-center text-xs uppercase tracking-widest text-gray-500 after:absolute after:top-full after:left-0 after:right-0 after:bottom-auto after:h-2 after:bg-gray-100 after:content-['']"
          :class="{
            'font-black text-gray-900 after:bg-[#318B12]': step.active
          }"
        >
          {{ step.label }}
        </span>
      </li>
    </ul>
  </div>
</template>

<script setup lang="ts">
import { defineProps, computed } from 'vue';
import type { ProgressBarStep } from './Tz-Progress-Bar.types';

const computedSteps = computed(() =>
  props.steps.map((x, i) => {
    return {
      active: i == props.selectedStepIndex,
      enabled: i < props.selectedStepIndex,
      label: x.label,
      url: x.url
    };
  })
);

const props = defineProps<{
  steps: ProgressBarStep[];
  selectedStepIndex: number;
}>();
</script>
