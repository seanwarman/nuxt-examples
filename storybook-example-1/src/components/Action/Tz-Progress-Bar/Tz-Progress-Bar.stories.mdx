import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';
import { withDesign } from 'storybook-addon-designs';
import { MobileViewports } from '@/helpers/viewports';

import { exampleSteps } from './Tz-Progress-Bar.data';
import TzProgressBar from './Tz-Progress-Bar.vue';

import vueRouter from 'storybook-vue3-router';

<Meta title="Components/Action/Progress Bar" component={TzProgressBar} decorators={[withDesign, vueRouter()]} />

export const Template = (args) => ({
  components: { TzProgressBar },
  setup() {
    return { args };
  },
  template: '<tz-progress-bar v-bind="args" />'
});

# Progress Bar

<Canvas withSource>
  <Story
    name="Default"
    args={{
      steps: [...exampleSteps],
      selectedStepIndex: 0
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Default" />

## Three Steps Example

<Canvas withSource isColumn>
  <Story
    name="Step 1"
    args={{
      steps: [...exampleSteps],
      selectedStepIndex: 0
    }}
  >
    {Template.bind({})}
  </Story>
  <Story
    name="Step 2"
    args={{
      steps: [...exampleSteps],
      selectedStepIndex: 1
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
  <Story
    name="Step 3"
    args={{
      steps: [...exampleSteps],
      selectedStepIndex: 2
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
