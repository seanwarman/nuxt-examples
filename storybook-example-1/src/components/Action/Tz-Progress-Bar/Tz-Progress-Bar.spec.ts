/**
 * @vitest-environment jsdom
 */

import { mount } from '@vue/test-utils';
import TzProgressBar from './Tz-Progress-Bar.vue';
import { expect, test } from 'vitest';
import { exampleSteps } from './Tz-Progress-Bar.data';

const wrapper = mount(TzProgressBar, {
  props: {
    steps: exampleSteps,
    selectedStepIndex: 0
  }
});

test('renders correctly', () => {
  expect(wrapper.element).toMatchSnapshot();
});
