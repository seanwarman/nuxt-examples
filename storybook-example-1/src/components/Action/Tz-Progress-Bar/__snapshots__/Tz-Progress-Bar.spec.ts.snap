// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  class="m-auto w-full pb-2 max-w-[395px]"
>
  <ul
    class="flex w-full"
  >
    
    <li
      class="relative flex-1 active"
    >
      <span
        class="inline-block h-full w-full p-2 text-center text-xs uppercase tracking-widest text-gray-500 after:absolute after:top-full after:left-0 after:right-0 after:bottom-auto after:h-2 after:bg-gray-100 after:content-[''] font-black text-gray-900 after:bg-[#318B12]"
      >
        Your details
      </span>
    </li>
    <li
      class="relative flex-1"
    >
      <span
        class="inline-block h-full w-full p-2 text-center text-xs uppercase tracking-widest text-gray-500 after:absolute after:top-full after:left-0 after:right-0 after:bottom-auto after:h-2 after:bg-gray-100 after:content-['']"
      >
        Delivery
      </span>
    </li>
    <li
      class="relative flex-1"
    >
      <span
        class="inline-block h-full w-full p-2 text-center text-xs uppercase tracking-widest text-gray-500 after:absolute after:top-full after:left-0 after:right-0 after:bottom-auto after:h-2 after:bg-gray-100 after:content-['']"
      >
        Payment
      </span>
    </li>
    
  </ul>
</div>
`;
